assets/js/autocomplete.ts

```diff
@@ -232,29 +115,33 @@ function toggleSearchAutocomplete() {
 }
 
 function listenAutocomplete() {
-  let timeout: number | undefined;
+  let serverSideSuggestionsTimeout: number | undefined;
 
   let localAc: LocalAutocompleter | null = null;
   let localFetched = false;
 
   document.addEventListener('focusin', fetchLocalAutocomplete);
 
   document.addEventListener('input', event => {
-    removeParent();
+    popup.hide();
     fetchLocalAutocomplete(event);
```

Comment:
Fetching local autocomplete results maybe should also be moved into the suggestions file

Reply:
Do you mean the `fetchLocalAutocomplete` function or the whole logic of receiving `input` events from inputs and then returning just a list of suggestions to render?

Reply:
I think just the fetchLocalAutocomplete function for now... I am not sure if I can clearly articulate the separations of concerns in just a comment, I can follow up myself with that

Reply:
Alrighty!